src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -63,19 +67,75 @@ public ConfigXml(Context context) throws OpenConfigException {
             generateKeysConfig(context);
         }
 
-        try {
-            DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-            mConfig = db.parse(mConfigFile);
-        } catch (SAXException | ParserConfigurationException | IOException e) {
-            throw new OpenConfigException();
-        }
+	readConfig();
 
         if (isFirstStart) {
             changeLocalDeviceName();
             changeDefaultFolder();
         }
     }
 
+    private void readConfig() {
+	if (!mConfigFile.canRead() && !checkPermConfigFile()) {
+	    throw new OpenConfigException();
+	}
+        try {
+            DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+	    Log.d(TAG,"Trying to read '" + mConfigFile + "'");
+            mConfig = db.parse(mConfigFile);
+        } catch (SAXException | ParserConfigurationException | IOException e) {
+	  Log.w(TAG,"Cannot read '" + mConfigFile + "'",e);
+	  throw new OpenConfigException();
+	}
+    }
+
+    private boolean checkPermConfigFile() {
```

Comment:
Please add some javadoc to describe what the method does, as its relatively complicated. You can probably move some of your inline comments into javadoc.